### DISCUSSION TITLE
Form in Display mode / disable all fields and buttons in the form

### DISCUSSION THREAD

**Discussion by tanetrail** [react-hook-form/react-hook-form#4312] [Q&A] [answered]
Is there a way to disable all fields and buttons in the form by default and enable it on button click. want to see if we can show the form in the display mode and later change it to edit mode. tried with fieldset but select/checkbox are not working as expected

**Comment by jorisre** [comment]
Hi @tanetrail 

Could you provide an example please ?

It works for me with simple React input, select, checkbox etc..: https://codesandbox.io/s/friendly-http-07rw4

**Reply by tanetrail** [reply]
https://stackblitz.com/edit/ionic-react-hook-form-eisv24?file=src%2FApp.tsx

Only text field is disabled, i can change the remaining fields

**Reply by jorisre** [reply]
It doesn't work with the `fieldset` because Ionic Select and Checkbox aren't native input. If you want to disable them, you've to pass the `disabled` props yourself:

```ts
<IonCheckbox
    disabled // 👈🏻 
    slot="start"
    checked={value}
    onIonChange={(e: CustomEvent<CheckboxChangeEventDetail>) => {
       onChange(e?.detail.checked);
       console.log(e);
     }}
/>
```

**Comment by leapful** [comment, marked as answer]
@tanetrail @jorisre 

Since Ionic React is built from Web Component, disabled state can be changed directly inside ion-item which is the parent wrapper of all Ionic components via attribute "disabled". We can use a custom Fieldset to toggle disabled attribute.

For example:
```
import React from "react";

const Fieldset = ({ children, ...props }) => {
  const { disabled } = props;
  const ref = React.useRef(null);

  React.useEffect(() => {
    if (ref.current) {
      const items: HTMLElement[] = Array.from(
        ref.current.querySelectorAll("ion-item")
      );

      for (const item of items) {
        if (disabled) {
          item.setAttribute("disabled", "true");
        } else {
          item.removeAttribute("disabled");
        }
      }
    }
  }, [disabled, ref.current]);

  return (
    <fieldset ref={ref} disabled={disabled}>
      {children}
    </fieldset>
  );
};

export default Fieldset;
```

Working sample: https://stackblitz.com/edit/ionic-react-hook-form-icwatk?file=src%2FApp.tsx
